src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -413,7 +413,7 @@ namespace HeatBalanceSurfaceManager {
                     Surface(SurfNum).WindDir = Node(Surface(SurfNum).LinkedOutAirNode).OutAirWindDir;
                 }
 
-                if (Surface(SurfNum).HasSurroundingSurfProperties) {
+                if (InitSurfaceHeatBalancefirstTime && Surface(SurfNum).HasSurroundingSurfProperties) {
```

Comment:
This looks fine.  This would just be to avoid wasting runtime, right?

Reply:
I actually don't like seeing things like this because each time through must evaluate 2 expressions. I'd rather see:

    if (InitSurfaceHeatBalancefirstTime) {
        if (whatever) {
            ... do something
            InitSurfaceHeatBalancefirstTime = false;
        }
    }

Reply:
Wait a minute. Does this need to be InitSurfaceHeatBalancefirstTime(SurfNum) ??

Reply:
@rraustad you're looking for bonus points.  

From a mental perspective, it does look good your way because you don't have to mentally process two different things as you are understanding the code.  From a code perspective however, as long as we are using the built-in boolean operators, the operator will "short-circuit", and not actually evaluate the second one if it doesn't need to.  In this case, if `InitSurfaceHeatBalancefirstTime` is false, it won't attempt to evaluate the second one because it knows it doesn't need to.  

There are more complex situations where you might have to write your own operators.  Perhaps if you wanted to compare two class instances in a very custom way, you would write your own operator.  In this case, whether it short-circuits or not is up to your own implementation.

Reply:
> Wait a minute. Does this need to be InitSurfaceHeatBalancefirstTime(SurfNum) ??

I thought so too!  But I don't think it does.  It is right inside of a tight SurfNum loop, so it looks to be valid for all surfaces, one time through.

Reply:
Good explanation on the short-circuit. Compilers are smarter than I am. Maybe I need to rethink my strategy.

Reply:
I had to take a small detour just for fun.  This demonstrates a small example of a custom operator.  Let's say you have a Person class that defines a bunch of attributes, but you only care about whether they are awesome.  Your custom && operator can just compare that one thing, and you can do short-circuiting pretty easily.  OK I'm heading back to work slash tornado preparation.

```
#include <iostream>

struct Person {
    bool awesome = false;
    bool operator&&(const Person &x) const
    {
        // try to short-circuit
        if (!this->awesome) {
            std::cout << "SHORT CIRCUITING\n";
            return false;
        }
        // only check the second arg if necessary
        std::cout << "COULD NOT SHORT CIRCUIT\n";
        return x.awesome;
    }
};

int main() {
    Person awesomePerson {true};
    Person notAwesomePerson {false};
    std::cout << "Comparing awesome person first -- should need to check both\n";
    if (awesomePerson && notAwesomePerson) {}
    std::cout << "Comparing not-awesome person first -- should short circuit\n";
    if (notAwesomePerson && awesomePerson) {}
}

// running this results in:
// Comparing awesome person first -- should need to check both
// COULD NOT SHORT CIRCUIT
// Comparing not-awesome person first -- should short circuit
// SHORT CIRCUITING
```

Reply:
> Wait a minute. Does this need to be InitSurfaceHeatBalancefirstTime(SurfNum) ??
Not sure I quite understand it. It should only be calculated once, but I guess not necessarily here. It only has to come after surface geometry calculation after `ViewFactorSkyIR` and `ViewFactorSkyGround` are calculated if not by user input. Perhaps at the end of  GetInputData?

Reply:
It's just that we are trained to be on alert for certain patterns. When checking a global flag (`InitSurfaceHeatBalancefirstTime`) and then a "per-surface" condition (`Surf(SurfNum).a`), together, it is a clue that this loop could potentially only be run for one individual surface.  In this case, the loop is tightly packed into this single function, so I think it is fine.  But we're always watching for that sort of thing.

Reply:
@Myoldmopar is awesomePerson! But Person is a custom operator, still I think you proved your point.

Reply:
The true test is for the x10 team where the E+ function with highest call count is tested with this hierarchy. A single if test with nested if vs multiple argument if test to see impact of compiler short-circuiting (with first argument as most likely false). Thanks for the home schooling.